featurebyte/query_graph/sql/specs.py

```diff
@@ -698,6 +701,47 @@ def construct_specs(
         return specs
 
 
+@dataclass
+class ForwardAggregateSpec(NonTileBasedAggregationSpec):
+    """
+    ForwardAggregateSpec contains all information required to generate sql for a forward aggregate target.
+    """
+
+    parameters: ForwardAggregateParameters
+
+    def agg_result_name(self) -> str:
+        return self.get_agg_result_name_from_groupby_parameters(self.parameters)
+
+    @property
+    def aggregation_type(self) -> AggregationType:
+        return AggregationType.FORWARD
+
+    def get_source_hash_parameters(self) -> dict[str, Any]:
+        # TODO:
```

Comment:
wasn't sure what else needed to be added here

Reply:
For now this can follow what was done for item aggregate: https://github.com/featurebyte/featurebyte/blob/5fab5ef8cb45b1bd94686ecf046fcc5d551f5e5b/featurebyte/query_graph/sql/specs.py#L527-L535

The parameters here are used to determine the specs that can be processed in the same subquery. Should be possible to refactor and avoid duplicated logic later.